### ISSUE TITLE
FEAT: AWS Athena Supsport

### ISSUE THREAD

**Issue by hoffrocket** [beekeeper-studio/beekeeper-studio#462]
I'd like to be able to use Beekeeper with AWS Athena. I've attempted to implement myself, but hit a bit of a wall in using the `aws-sdk` library as webpack seems to include the browser target rather than the node target.

Any tips or guidance? I'm no webpack expert and am mostly cargo culting here, but I've already tried marking the library as `external`:  https://github.com/hoffrocket/beekeeper-studio/blob/athena/vue.config.js#L12

I can drive this forward and send a PR.

**Comment by rathboma** [comment]
👋 Hey Jon :-). Good to see you!

I'll take a look. Webpack is the worst with this type of stuff.

**Comment by rathboma** [comment]
How do you know webpack is including the browser target?

**Comment by hoffrocket** [comment]
I can infer it because the defaultProviders is empty here: https://github.com/hoffrocket/beekeeper-studio/blob/athena/src/lib/db/clients/athena.js#L25

If the node target were used that would be populated using the `node_loader` from the source here:
https://github.com/aws/aws-sdk-js/blob/master/lib/node_loader.js#L62

**Comment by rathboma** [comment]
Ok, so this is probably an electron quirk. I took a look at your code and what you linked. Here's what I think will fix it:

1. I don't think that means you're running the browser version. The renderer runs node code, but doesn't have access to the NODEJS environment variables, so the defaultProvider probably can't resolve the key and secret, or resolve the user home directory to load from the default config file. To test, try specifically using a `new AWS.Credentials` and specifying access key and secret key directly. 
  - If you really want it to read from the node env, you can use `electron.remote.process.env`. You can see I do this a lot in `platform_info.ts`
  - If you really wanted to inspect the defaultProvider in the `background.js` file -- that actually legit runs node, no browser, separate process.

2. You shouldn't have the aws-sdk as an external, either as an electron external, or as a webpack external. Webpack externals mean that the package is provided by the end system, similar to the `provided` tag in sbt/maven.

3. You should move `aws-sdk` from devDependencies to `dependencies`, devDependencies is really just for the build chain.

4. You might need to add `json-loader` to dev dependencies, then add a `loaders` section to the webpack config with this:

```js
    loaders: [
      {
        test: /\.json$/,
        loaders: ['json']
      }
    ],
```

I'm thinking as I'm typing, but I'm going to test this with my AWS account in a minute

**Comment by rathboma** [comment]
That seems to work, but you actually don't need the `loaders` stuff. I could connect and got a `query failed` error, which seems positive?

**Comment by hoffrocket** [comment]
Ah, so the issue is that I actually do want to use the provider chain so that the SDK picks up my session credentials that are already in my ~/.aws/credentials file

**Comment by rathboma** [comment]
Yes, you'll either have to load it on the background process, then send to the renderer at initialization, or bootstrap enough of the renderer env using electron remote.

If you get it reading credentials in a hacky way I can help fix it up and get it working right

**Comment by fhsgoncalves** [comment]
any news around the athena support?

**Comment by Mythirion** [comment]
Also would like to get Athena supported - connecting via the Web Console is fine and all but having access to all my DB connections in one place is a strong preference. Trying to drive using Beekeeper more around the business so the more compatible connections we can have, the better.
